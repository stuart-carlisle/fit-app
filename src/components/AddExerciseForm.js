import React, { useState , useEffect} from 'react'
import moment from 'moment'
import { connect } from 'react-redux'
import numeral from 'numeral'

export const AddExerciseForm = ({
  filters = {},
  onSubmit, 
  exercise = {
    description:'',
    type:'',
    energy:'',
    time:''
  }
}) => {
    const defaultEnergy = numeral(exercise.energy).value()
    const defaultTime = numeral(exercise.time).value()
    const date = filters.date
    const referenceDatabase = filters.referenceDatabase
    const [time, setTime] = useState(defaultTime)
    const [error, setError] = useState('')
    const [energy, setEnergy] = useState(defaultEnergy)
    const diaryDate = date ? date.valueOf() : moment().startOf('day').valueOf()
    const lastUsed = moment().valueOf()

    useEffect(()=>{
        
      let refreshed = JSON.parse(localStorage.getItem('refreshed'))
      if(refreshed){
        window.history.back()
        window.scrollTo(0,0)
        localStorage.removeItem('refreshed')
      }else{
        localStorage.setItem('refreshed',JSON.stringify(true))
      }
    },[])

    const onFormSubmit = (e) => {
        e.preventDefault();
        const exerciseData = { 
          description: exercise.description,
          type: exercise.type
        }
        if (numeral(time).value()<=0) {
          setError('Please provide a time greater than 0')
          window.scrollTo(0,0)
        } else {
          setError('')
          onSubmit(
            referenceDatabase,
            diaryDate,
            lastUsed,
            exerciseData,
            time.toString(),
            energy.toString()
          )
          localStorage.removeItem('refreshed')
        }
      }
    
    const onTimeChange = (e)=>{
        const time = e.target.value
          if (!time || time.match(/^\d{1,}(\.\d{0,1})?$/)) {
          setTime(time)
          const timeValue = numeral(time).value()
          setEnergy(Math.round(defaultEnergy * (timeValue/defaultTime)))
        }
    }

    const onEnergyChange = ()=>{
      
    }

    return (
        <div>
        <div className="content-container__background">
        <div className="content-container__form__update">
            <form className= "form" onSubmit={onFormSubmit}>
                {error && <p id="error-message" className="form__error">{error}</p>}
                  <h1 className="page-header__title-update"> {exercise.description} </h1>
                    <span className="text-input__span-container-form">
                      <div className="content-container__update-titles">
                        <h3 className="form__heading__update">Exercise Time (mins):</h3>
                        <h3 className="form__heading__update">Energy (calories): </h3>
                      </div>
                      <div className="content-container__update">
                        <input  //Exercise Duration
                          className="text-input__update-form"
                          type="text"
                          value={time}
                          onChange={onTimeChange}
                          placeholder="Enter a Duration"
                        />  
                        <input  //Energy Used
                          className="text-input__update-form"
                          type="text"
                          value={energy?energy:''}
                          onChange={onEnergyChange}
                          placeholder="Energy Used"
                        /> 
                      </div>
                    </span>
                <button className="button button--form">Add Exercise</button>
            </form>
          </div>
        </div>
      </div>  
    )
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  }
}


export default connect(mapStateToProps)(AddExerciseForm);