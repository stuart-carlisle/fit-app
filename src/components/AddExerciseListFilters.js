import React, { useState } from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAlphabetical, setReferenceDatabaseToPublic } from '../actions/filters';
import { setExercisesPublic } from '../actions/exercises'
import algoliasearch from 'algoliasearch/lite'

export const AddExerciseListFilters = ({
    filters,
    setTextFilter, 
    setReferenceDatabasePublic,
    sortByDate, 
    sortByAlphabetical,
    setExercisesPublic 
  }) => {
 
  const searchClient = algoliasearch (
      process.env.ALGOLIA_APP_ID,
      process.env.ALGOLIA_API_KEY
  )

  const index = searchClient.initIndex('exercises');

  const [text, setText] = useState('')
    
  const onTextChange = (e) => {
    setTextFilter(e.target.value)
    setText(e.target.value)
  }

  const onSortChange = (e) => {
    if (e.target.value === 'date') {
      sortByDate()
    } else if (e.target.value === 'alphabetical') {
      sortByAlphabetical()
    }
  }

  const onSubmit = (e) => {
    e.preventDefault()
    index
      .search({
        query: text
      })
      .then((responses) => {
        const array = responses.hits
        const newExercises = array.map((exercise)=>{
          return{
            description:exercise.description,
            type:exercise.type,
            energy:exercise.energy,
            time:exercise.time,
            id:exercise.objectID
          }
        })
        setExercisesPublic(newExercises)
      }).then(()=>{
        setReferenceDatabasePublic()
      })
}

    return (
      
      <div className="header__content--filters"> 
      <form className="filters-text-input__span-container" onSubmit={onSubmit}>
          <div className="input-group__item-page-header">
            <input
              type="text"
              className="filters-text-input"
              placeholder="SEARCH EXERCISES"
              value={text}
              onChange={onTextChange}
            />
          </div>
          <div className="input-group__item-page-header">
            <select
              className="minimal"
              value={filters.sortBy}
              onChange={onSortChange}
            >
              <option value="date">DATE</option>
              <option value="alphabetical">ALPHABETICAL</option>
            </select>
          </div>
            <button className="button filters-button--form-page-header">Search Online</button>
            </form>   
        </div>
    )
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAlphabetical: () => dispatch(sortByAlphabetical()),
  setReferenceDatabasePublic: () => dispatch(setReferenceDatabaseToPublic()),
  setExercisesPublic: (exercises) => dispatch(setExercisesPublic(exercises))
})

export default connect(mapStateToProps,mapDispatchToProps)(AddExerciseListFilters);

