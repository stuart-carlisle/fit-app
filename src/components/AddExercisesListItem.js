import React from 'react';
import { Link } from 'react-router-dom';

export const AddExercisesListItem = ({ description, energy, id }) => (


  <Link className="list-item" to={`/add-exercise-confirmation/${id}`}>
    <div>
      <p className="list-item__title">{description}</p>
    </div>
    <div>
      <p className="list-item__data">{energy}</p>
    </div>  
  </Link>

);

export default AddExercisesListItem;
