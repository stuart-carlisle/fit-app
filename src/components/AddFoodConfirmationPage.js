import React from 'react'
import { connect } from 'react-redux'
import AddFoodForm from './AddFoodForm'
import { Link } from 'react-router-dom'
import { startAddFoodDiary, startAddFoodPrivate, startUpdateFoodPrivate } from '../actions/foods'

export const AddFoodConfirmationPage = (props) => {
  
  const onSubmit = (referenceDatabase, diaryDate, lastUsed, numberOfServings, food, meal) => {
    if (referenceDatabase==='public'){
      props.startAddFoodPrivate({...food,lastUsed})
    } else if (referenceDatabase==='private'){
      props.startUpdateFoodPrivate(props.food.id,{...food,lastUsed})
    }
    props.startAddFoodDiary({...food, diaryDate, numberOfServings, meal})
    props.history.push('/diary');
  }
  

  return (
     <div>
     <Link className="button__back-button" to={`/add`}><img className="image__header" src="/images/back-arrow.svg"/></Link>
        <div className="page-header__filter page-header__filter--with-back">
          <div className="content-container">
            <h1 className="page-header__title">ADD FOOD</h1>
          </div>
        </div>
        <div className="show-for-mobile page-header__margin"></div>
        <div className="content-container">
        <AddFoodForm id="add-food-form" targets={props.targets} food={props.food} onSubmit={onSubmit} />
        </div>
     </div>
 )
}

const mapStateToProps = (state,props) => {
  if(state.filters.referenceDatabase==='public'){
    return {
      food: state.foodsPublic.find((food) => food.id === props.match.params.id),
      targets: state.targets
    }
  }else if(state.filters.referenceDatabase==='private'){
    return {
      food: state.foodsPrivate.find((food) => food.id === props.match.params.id),
      targets: state.targets
    }
  }
}

const mapDispatchToProps = (dispatch) =>({
    startAddFoodDiary: (food) => dispatch(startAddFoodDiary(food)),
    startUpdateFoodPrivate: (id, food) => dispatch(startUpdateFoodPrivate(id, food)),
    startAddFoodPrivate: (food) => dispatch(startAddFoodPrivate(food))
})

export default connect(mapStateToProps, mapDispatchToProps)(AddFoodConfirmationPage);




