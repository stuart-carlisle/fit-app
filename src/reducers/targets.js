// targets Reducer

const targetsReducerDefaultState = {
  dailyEnergyTarget: '2400',
  dailyCarbsTarget: '400',
  dailyFatTarget: '70'
}

export default (state = targetsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_DAILY_ENERGY_TARGET':
      return{
        ...state,
        dailyEnergyTarget: action.dailyEnergyTarget
      }
    case 'UPDATE_TARGETS':
      return{
        ...state,
        ...action.updates
      }
    case 'SET_TARGETS':
      return action.targets
    default:
      return state;
  }
};
