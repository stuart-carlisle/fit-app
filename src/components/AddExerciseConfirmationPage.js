import React from 'react'
import { connect } from 'react-redux'
import AddExerciseForm from './AddExerciseForm'
import { Link } from 'react-router-dom'
import { startAddExerciseDiary, startAddExercisePrivate, startUpdateExercisePrivate } from '../actions/exercises'

export const AddExerciseConfirmationPage = (props) => {
  
  const onSubmit = (referenceDatabase, diaryDate, lastUsed, exerciseDescAndType, time, energy) => {
    if (referenceDatabase==='public'){
      props.startAddExercisePrivate({...exerciseDescAndType, time, energy, lastUsed})
    } else if (referenceDatabase==='private'){
      props.startUpdateExercisePrivate(props.exercise.id,{...exerciseDescAndType, time, energy, lastUsed})
    }
    props.startAddExerciseDiary({...exerciseDescAndType, time, diaryDate, energy})
    props.history.push('/diary');
  }
  

  return (
     <div>
     <Link className="button__back-button" to={`/add-exercise`}><img className="image__header" src="/images/back-arrow.svg"/></Link>
        <div className="page-header__filter page-header__filter--with-back">
          <div className="content-container">
            <h1 className="page-header__title">ADD EXERCISE</h1>
          </div>
        </div>
        <div className="show-for-mobile page-header__margin"></div>
        <div className="content-container">
        <AddExerciseForm id="add-exercise-form" exercise={props.exercise} onSubmit={onSubmit} />
        </div>
     </div>
 )
}

const mapStateToProps = (state,props) => {
  if(state.filters.referenceDatabase==='public'){
    return {
      exercise: state.exercisesPublic.find((exercise) => exercise.id === props.match.params.id)
    }
  }else if(state.filters.referenceDatabase==='private'){
    return {
      exercise: state.exercisesPrivate.find((exercise) => exercise.id === props.match.params.id)
    }
  }
}

const mapDispatchToProps = (dispatch) =>({
    startAddExerciseDiary: (exercise) => dispatch(startAddExerciseDiary(exercise)),
    startUpdateExercisePrivate: (id, exercise) => dispatch(startUpdateExercisePrivate(id, exercise)),
    startAddExercisePrivate: (exercise) => dispatch(startAddExercisePrivate(exercise))
})

export default connect(mapStateToProps, mapDispatchToProps)(AddExerciseConfirmationPage);




