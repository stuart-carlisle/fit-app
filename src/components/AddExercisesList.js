import React from 'react'
import { connect } from 'react-redux'
import AddExercisesListItem from './AddExercisesListItem'
import selectExercises from '../selectors/exercises'

export const AddExercisesList = ({exercises}) => {
  let exercisesTopResults
  
  if(exercises.length>10){
    exercisesTopResults = exercises.slice(0,10)
  }else{
    exercisesTopResults = exercises
  }

  return(<div className="content-container">
   <div className="list-header">
    <div className="">Exercise</div>
    <div className="show-for-tablet">Calories Burned</div>
    <div className="show-for-mobile">Cals Burned</div>
   </div>
    <div className="list-body">
      {
        exercisesTopResults.length === 0 ? (
          <div className="list-item list-item--message">
            <span>No Exercises In History</span>
          </div>
        ) : (
         exercisesTopResults.map((exercise) => {
            return <AddExercisesListItem key={exercise.id} {...exercise} />
          })
        )
      }
    </div>
  </div>  
  )
}

const mapStateToProps = (state) => {
  return {
    exercises: state.filters.referenceDatabase === 'public' ? state.exercisesPublic : selectExercises(state.exercisesPrivate,  state.filters)
  }
}

export default connect(mapStateToProps)(AddExercisesList);
