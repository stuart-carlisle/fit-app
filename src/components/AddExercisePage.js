import React, { useEffect } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import AddExerciseListFilters from './AddExerciseListFilters'
import AddExercisesList from './AddExercisesList'
import { setTextFilter, sortByDate, setReferenceDatabaseToPrivate } from '../actions/filters'

export const AddExercisePage = ({ setTextFilter, sortByDate, setReferenceDatabaseToPrivate}) => {
  useEffect(()=>{
    localStorage.removeItem('refreshed')
      setTextFilter('')
      sortByDate()
      setReferenceDatabaseToPrivate()
  },[])
    return(
      <div>
      <Link className="button__back-button" to={`/diary`}><img className="image__header" src="/images/back-arrow.svg"/></Link>
    <div className="page-header__filter page-header__filter--with-back">
      <div className="content-container">
        <AddExerciseListFilters />
      </div>  
    </div>
    <AddExercisesList />
    <div className="content-container">
    <Link className="button button--link" to={`/create-exercise`}>+ CREATE CUSTOM EXERCISE</Link>
    </div>
    </div>
    )
}

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  setReferenceDatabaseToPrivate: () => dispatch(setReferenceDatabaseToPrivate())
})

export default connect (undefined, mapDispatchToProps)(AddExercisePage)