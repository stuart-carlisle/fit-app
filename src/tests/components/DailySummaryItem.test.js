import React from 'react'
import { mount, shallow } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import { DailySummaryItem  } from '../../components/DailySummaryItem'
import targets from '../fixtures/targets'
import moment from 'moment'

test('Should render dialy summary item correctly',()=>{
    const dailyTotal = 640
    const exerciseDailyTotal = 340
    const date = moment().subtract(1,'days').startOf('day')
    const setDateFilter = jest.fn()

    const wrapper = shallow(<DailySummaryItem 
            setDateFilter={setDateFilter}
            dailyTotal={dailyTotal}
            exerciseDailyTotal={exerciseDailyTotal}
            date={date}
            targets={targets}
        />)
    expect(wrapper).toMatchSnapshot()
})


test('Should render display date correctly',()=>{
    const dailyTotal = 640
    const exerciseDailyTotal = 340
    const date = moment().subtract(1,'days').startOf('day')
    const setDateFilter = jest.fn()
    const displayDate = 'YESTERDAY'

    const wrapper = mount(
        <MemoryRouter>
        <DailySummaryItem
            setDateFilter={setDateFilter}
            dailyTotal={dailyTotal}
            exerciseDailyTotal={exerciseDailyTotal}
            date={date}
            targets={targets}
        />
        </MemoryRouter>
    )
    expect(wrapper.find('.show-for-mobile').text()).toBe(displayDate)
    expect(wrapper.find('.show-for-tablet').text()).toBe(displayDate)

    const date2 = moment().startOf('day')
    const displayDate2 = 'TODAY'

    const wrapper2 = mount(
        <MemoryRouter>
        <DailySummaryItem
            setDateFilter={setDateFilter}
            dailyTotal={dailyTotal}
            exerciseDailyTotal={exerciseDailyTotal}
            date={date2}
            targets={targets}
        />
        </MemoryRouter>
    )
    expect(wrapper2.find('.show-for-tablet').text()).toBe(displayDate2)
    expect(wrapper2.find('.show-for-mobile').text()).toBe(displayDate2)

    const date3 = moment().subtract(2,'days').startOf('day')
    const displayDate3 = '30th Dec'

    const wrapper3 = mount(
        <MemoryRouter>
        <DailySummaryItem
            setDateFilter={setDateFilter}
            dailyTotal={dailyTotal}
            exerciseDailyTotal={exerciseDailyTotal}
            date={date3}
            targets={targets}
        />
        </MemoryRouter>
    )
    expect(wrapper3.find('.show-for-tablet').text()).toBe(displayDate3)
    expect(wrapper3.find('.show-for-mobile').text()).toBe(displayDate3)
})

test('Should render values correctly on the dialy summary item',()=>{
    const dailyTotal = 640.23
    const exerciseDailyTotal = 340.82
    const date = moment().subtract(1,'days').startOf('day')
    const setDateFilter = jest.fn()

    const wrapper = mount(
        <MemoryRouter>
            <DailySummaryItem 
                setDateFilter={setDateFilter}
                dailyTotal={dailyTotal}
                exerciseDailyTotal={exerciseDailyTotal}
                date={date}
                targets={targets}
            />
        </MemoryRouter>
    )
    expect(wrapper.find('#target').text()).toBe('2000')
    expect(wrapper.find('#daily-total').text()).toBe('640')
    expect(wrapper.find('#exercise-daily-total').text()).toBe('341')
    expect(wrapper.find('#difference').text()).toBe('1701')
})


test('Should call setDateFilter with date', ()=>{
    const dailyTotal = 640.23
    const exerciseDailyTotal = 340.82
    const date = moment().subtract(1,'days').startOf('day')
    const setDateFilter = jest.fn()

    const wrapper = mount(
        <MemoryRouter>
            <DailySummaryItem 
                setDateFilter={setDateFilter}
                dailyTotal={dailyTotal}
                exerciseDailyTotal={exerciseDailyTotal}
                date={date}
                targets={targets}
            />
        </MemoryRouter>
    )

    wrapper.find('.header__content2').simulate('click',{
    })
    expect(setDateFilter).toHaveBeenCalledWith(date)
})




