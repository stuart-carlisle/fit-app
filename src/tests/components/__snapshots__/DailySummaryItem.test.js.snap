// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render dialy summary item correctly 1`] = `
<div
  className="header__content2"
  onClick={[Function]}
>
  <Link
    className="summary"
    to="/diary"
  >
    <div
      className="loader__fade"
    >
      <div
        className="summary-text__mobile-title show-for-mobile"
      >
        31st December
      </div>
      <span
        className="summary-text__date summary-item--span-dashboard"
      >
        <div
          className="summary-text__date show-for-tablet"
        >
          31st December
        </div>
        <div>
          <span
            className="summary-item--span"
          >
            <div>
              <span
                className="summary-item--span"
              >
                <div
                  className="summary-text__container"
                >
                  <div
                    className="summary-text"
                    id="target"
                  >
                    0
                  </div>
                  <div
                    className="summary-text__small blue"
                  >
                    target
                  </div>
                </div>
                <div
                  className="left-space"
                >
                  -
                </div>
              </span>
            </div>
            <div>
              <span
                className="summary-item--span"
              >
                <div
                  className="summary-text__container red left-space"
                >
                  <div
                    className="summary-text"
                    id="daily-total"
                  >
                    640
                  </div>
                  <div
                    className="summary-text__small blue"
                  >
                    food
                  </div>
                </div>
                <div
                  className="left-space"
                >
                  +
                </div>
              </span>
            </div>
            <div>
              <span
                className="summary-item--span"
              >
                <div
                  className="summary-text__container green left-space"
                >
                  <div
                    className="summary-text"
                    id="exercise-daily-total"
                  >
                    340
                  </div>
                  <div
                    className="summary-text__small blue"
                  >
                    exercise
                  </div>
                </div>
                <div
                  className="left-space"
                >
                  =
                </div>
              </span>
            </div>
            <div
              className="summary-text__container red left-space"
            >
              <div
                className="summary-text"
                id="difference"
              >
                -300
              </div>
              <div
                className="summary-text__small blue"
              >
                total
              </div>
            </div>
          </span>
        </div>
      </span>
      <MultiColorProgressBarContainer
        data={
          Object {
            "combinedTotal": 300,
            "target": 0,
          }
        }
      />
    </div>
  </Link>
</div>
`;
